numpy/core/defchararray.py

```diff
@@ -2206,7 +2204,7 @@ def __radd__(self, other):
         --------
         add
         """
-        return asarray(add(numpy.asarray(other), self))
+        return add(other, self)
```

Comment:
Fun part, we effectively allow any dtype now in the `char` functions :).

Reply:
Maybe, but I don't really care.  The only expectation that breaks tis that someone might think it casts to string, but it never did (it only ever failed in various ways).
And arguably it is easy to detect a numpy number, but maybe that easy to say "it's definitely a string".